src/ax/routing/dispatcher.go

```diff
@@ -3,14 +3,16 @@ package routing
 import (
 	"context"
 
-	"github.com/jmalloc/ax/src/ax"
 	"github.com/jmalloc/ax/src/ax/endpoint"
+
+	"github.com/jmalloc/ax/src/ax"
 )
 
 // Dispatcher is an inbound pipeline stage that routes messages to the
 // appropriate MessageHandler instances according to a "handler table".
 type Dispatcher struct {
-	Routes HandlerTable
+	Routes     HandlerTable
+	Validators []endpoint.Validator
```

Comment:
This validators slice should be unexposed, and populated from the validators in the endpoint when `d.Initialize()` is called.

Reply:
No worries, will move the validators to the endpoint. Thanks for clarifying that as I was obviously not going to change `Initialize()` method to pass the validators. Will commit the changes soon ;-)